File: bms_app/lib/models/transport/client_req_dto.dart
class ClientSignInRequest {
  const ClientSignInRequest({required this.identifier, required this.password});

  final String identifier;
  final String password;
}

class ClientRefreshSessionRequest {
  const ClientRefreshSessionRequest({
    required this.sessionId,
    required this.refreshToken,
    required this.tokenId,
    required this.tokenFamilyId,
    required this.principalId,
    this.scopes = const [],
  });

  final String sessionId;
  final String refreshToken;
  final String tokenId;
  final String tokenFamilyId;
  final String principalId;
  final List<String> scopes;
}

class ClientAuthCredentialsResponse {
  const ClientAuthCredentialsResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.downstreamToken,
    required this.tokenType,
    required this.sessionId,
    required this.tokenId,
    required this.principalId,
    required this.tokenFamilyId,
    required this.scopes,
    required this.issuedAtMs,
    required this.accessExpiresAtMs,
    required this.refreshExpiresAtMs,
    required this.persisted,
  });

  final String accessToken;
  final String refreshToken;
  final String downstreamToken;
  final String tokenType;
  final String sessionId;
  final String tokenId;
  final String principalId;
  final String tokenFamilyId;
  final List<String> scopes;
  final int issuedAtMs;
  final int accessExpiresAtMs;
  final int refreshExpiresAtMs;
  final bool persisted;
}

class ClientUserProfileRequest {
  const ClientUserProfileRequest({required this.identifier});

  final String identifier;
}

class ClientRegisterRequest {
  const ClientRegisterRequest({
    required this.username,
    required this.password,
    this.email = '',
    this.phone = '',
  });

  final String username;
  final String email;
  final String phone;
  final String password;
}

class ClientHomeSnapshotRequest {
  const ClientHomeSnapshotRequest({this.deviceId});

  final String? deviceId;
}

class ClientRecordStationOptionsRequest {
  const ClientRecordStationOptionsRequest({this.includeOffline});

  final bool? includeOffline;
}

class ClientRecordsCursorRequest {
  const ClientRecordsCursorRequest({
    this.startAtMs,
    this.endAtMs,
    this.deviceId,
    this.keyword,
    this.confidenceMin,
    this.cursor,
    this.limit = 20,
    this.sort = 'captured_at_ms_desc',
  });

  final int? startAtMs;
  final int? endAtMs;
  final String? deviceId;
  final String? keyword;
  final double? confidenceMin;
  final String? cursor;
  final int limit;
  final String sort;
}

// ── Chat request DTOs ──────────────────────────────────────────────

class ChatImageRef {
  const ChatImageRef({
    this.imageId = '',
    this.data = '',
    this.mimeType = 'image/jpeg',
    this.filename = '',
  });

  final String imageId;
  final String data;
  final String mimeType;
  final String filename;
}

class ChatSendRequest {
  const ChatSendRequest({
    required this.sessionId,
    required this.userId,
    required this.text,
    this.images = const [],
    this.locale = 'zh-CN',
    this.timezone = 'Asia/Shanghai',
    this.traceId = '',
  });

  final String sessionId;
  final String userId;
  final String text;
  final List<ChatImageRef> images;
  final String locale;
  final String timezone;
  final String traceId;
}

class ChatSessionListRequest {
  const ChatSessionListRequest({
    required this.userId,
    this.limit = 20,
    this.offset = 0,
  });

  final String userId;
  final int limit;
  final int offset;
}

class ChatSessionGetRequest {
  const ChatSessionGetRequest({
    required this.sessionId,
    required this.userId,
    this.messageLimit = 50,
  });

  final String sessionId;
  final String userId;
  final int messageLimit;
}

class ChatSessionDeleteRequest {
  const ChatSessionDeleteRequest({
    required this.sessionId,
    required this.userId,
  });

  final String sessionId;
  final String userId;
}

class ChatSessionCreateRequest {
  const ChatSessionCreateRequest({
    required this.userId,
    this.title = '',
  });

  final String userId;
  final String title;
}

class ClientWeeklyTrendRequest {
  const ClientWeeklyTrendRequest({this.days = 7, this.deviceId});

  final int days;
  final String? deviceId;
}

class ClientRangeSummaryRequest {
  const ClientRangeSummaryRequest({
    required this.startAtMs,
    required this.endAtMs,
    this.deviceId,
  });

  final int startAtMs;
  final int endAtMs;
  final String? deviceId;
}

File: data_server/src/models/business/client_req_dto.py
from __future__ import annotations

from typing import Literal

from msgspec import Struct


class ClientUserProfileRequest(Struct, kw_only=True):
	"""用户资料查询请求模型。

	用于客户端在登录成功后，根据登录输入的标识（用户名/邮箱/手机号）拉取用户资料。
	"""

	identifier: str = ""


class ClientRegisterRequest(Struct, kw_only=True):
	"""注册请求模型。

	用于 `POST /v1/client/users/register`，承载注册页提交的用户名、可选邮箱、可选手机号和密码。
	"""

	username: str = ""
	email: str = ""
	phone: str = ""
	password: str = ""


class ClientHomeSnapshotRequest(Struct, kw_only=True):
	"""首页概览请求模型。

	用于 `GET /v1/client/home/summary`，当前仅保留 `device_id` 作为可选过滤/观测字段。
	"""

	device_id: str | None = None


class ClientRecordStationOptionsRequest(Struct, kw_only=True):
	"""记录页站点选项请求模型。

	用于 `GET /v1/client/records/stations`，可控制是否包含离线站点。
	"""

	include_offline: bool | None = None


class ClientRecordsCursorRequest(Struct, kw_only=True):
	"""记录列表游标请求模型。

	用于 `GET /v1/client/records`，承载时间范围、站点筛选与游标参数。
	"""

	start_at_ms: int | None = None
	end_at_ms: int | None = None
	device_id: str | None = None
	keyword: str | None = None
	confidence_min: float | None = None
	cursor: str | None = None
	limit: int = 20
	sort: Literal["captured_at_ms_desc"] = "captured_at_ms_desc"


class ClientWeeklyTrendRequest(Struct, kw_only=True):
	"""最近七日趋势请求模型。

	用于 `GET /v1/client/stats/weekly-trend`，可指定天数和设备筛选条件。
	"""

	days: int = 7
	device_id: str | None = None


class ClientRangeSummaryRequest(Struct, kw_only=True):
	"""时间段统计请求模型。

	用于 `GET /v1/client/stats/range-summary`，承载时间区间和可选设备筛选条件。
	"""

	start_at_ms: int = 0
	end_at_ms: int = 0
	device_id: str | None = None
